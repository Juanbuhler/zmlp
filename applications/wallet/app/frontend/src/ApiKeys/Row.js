import PropTypes from 'prop-types'

const DataQueueRow = ({ apiKey: { name, permissions } }) => {
  return (
    <tr>
      <td>{name}</td>
      <td>
        {permissions.join(', ').replace(/([A-Z])/g, match => ` ${match}`)}
      </td>
    </tr>
  )
}

DataQueueRow.propTypes = {
  apiKey: PropTypes.shape({
    name: PropTypes.string.isRequired,
    permissions: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  }).isRequired,
}

export default DataQueueRow
