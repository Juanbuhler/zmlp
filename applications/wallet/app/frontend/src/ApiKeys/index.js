import Head from 'next/head'
import { useRouter } from 'next/router'

import PageTitle from '../PageTitle'
import Tabs from '../Tabs'
import Table from '../Table'

import ApiKeysRow from './Row'

const ApiKeys = () => {
  const {
    query: { projectId },
  } = useRouter()

  return (
    <div>
      <Head>
        <title>API Keys</title>
      </Head>

      <PageTitle>Project API Keys</PageTitle>

      <Tabs
        tabs={[
          { title: 'View all', href: '/api-keys' },
          { title: 'Create API key', href: '/api-keys/add' },
        ]}
      />

      <Table
        url={`/api/v1/projects/${projectId}/apikeys/`}
        columns={['API Key Name', 'Permissions']}
        renderRow={({ result }) => (
          <ApiKeysRow key={result.id} apiKey={result} />
        )}
      />
    </div>
  )
}

export default ApiKeys
