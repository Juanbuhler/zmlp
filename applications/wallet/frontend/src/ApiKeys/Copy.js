import { constants, colors } from '../Styles'

const ApiKeysCopy = () => {
  return (
    <p
      css={{
        maxWidth: constants.paragraph.maxWidth,
        color: colors.structure.zinc,
      }}
    >
      Create and manage API Keys for applications or tools you would like to
      integrate with ZVI. An API Key is required to access data for this project
      using the ZVI SDK and REST APIs. For more information please visit{' '}
      <a
        css={{ color: colors.key.one }}
        target="_blank"
        rel="noopener noreferrer"
        href="https://zorroa.gitbook.io/zmlp/"
      >
        our documentation
      </a>
      .
    </p>
  )
}

export default ApiKeysCopy
