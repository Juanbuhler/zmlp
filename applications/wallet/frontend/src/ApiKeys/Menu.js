import { useState } from 'react'
import PropTypes from 'prop-types'
import Router from 'next/router'

import { fetcher } from '../Fetch/helpers'

import Menu from '../Menu'
import Button, { VARIANTS } from '../Button'
import ButtonActions from '../Button/Actions'
import Modal from '../Modal'

const ApiKeysMenu = ({ projectId, apiKeyId, revalidate }) => {
  const [isDeleteModalOpen, setDeleteModalOpen] = useState(false)
  const [isDeleting, setIsDeleting] = useState(false)

  return (
    <Menu open="left" button={ButtonActions}>
      {({ onClick }) => (
        <div>
          <ul>
            <li>
              <>
                <Button
                  variant={VARIANTS.MENU_ITEM}
                  onClick={() => {
                    setDeleteModalOpen(true)
                  }}
                  isDisabled={false}
                >
                  Delete
                </Button>
                {isDeleteModalOpen && (
                  <Modal
                    title="Delete API Key"
                    message="Deleting this key cannot be undone."
                    action={isDeleting ? 'Deleting...' : 'Delete Permanently'}
                    onCancel={() => {
                      setDeleteModalOpen(false)

                      onClick()
                    }}
                    onConfirm={async () => {
                      setIsDeleting(true)

                      await fetcher(
                        `/api/v1/projects/${projectId}/api_keys/${apiKeyId}/`,
                        { method: 'DELETE' },
                      )

                      await revalidate()

                      Router.push(
                        '/[projectId]/api-keys?action=delete-apikey-success',
                        `/${projectId}/api-keys`,
                      )
                    }}
                  />
                )}
              </>
            </li>
          </ul>
        </div>
      )}
    </Menu>
  )
}

ApiKeysMenu.propTypes = {
  projectId: PropTypes.string.isRequired,
  apiKeyId: PropTypes.string.isRequired,
  revalidate: PropTypes.func.isRequired,
}

export default ApiKeysMenu
