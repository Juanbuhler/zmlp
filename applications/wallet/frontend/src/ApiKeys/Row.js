import PropTypes from 'prop-types'

import Pills from '../Pills'

import ApiKeysMenu from './Menu'

const ApiKeysRow = ({
  projectId,
  apiKey: { id: apiKeyId, name, permissions },
  revalidate,
}) => {
  return (
    <tr>
      <td>{name}</td>
      <td>
        <Pills>
          {permissions.sort((a, b) => {
            if (a.toLowerCase() < b.toLowerCase()) return -1
            return 1
          })}
        </Pills>
      </td>
      <td>
        <ApiKeysMenu
          projectId={projectId}
          apiKeyId={apiKeyId}
          name={name}
          revalidate={revalidate}
        />
      </td>
    </tr>
  )
}

ApiKeysRow.propTypes = {
  projectId: PropTypes.string.isRequired,
  apiKey: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    permissions: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  }).isRequired,
  revalidate: PropTypes.func.isRequired,
}

export default ApiKeysRow
