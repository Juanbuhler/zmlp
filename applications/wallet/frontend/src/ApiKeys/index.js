import Head from 'next/head'
import { useRouter } from 'next/router'

import PageTitle from '../PageTitle'
import Tabs from '../Tabs'
import Table from '../Table'

import ApiKeysCopy from './Copy'
import ApiKeysRow from './Row'

const ApiKeys = () => {
  const {
    query: { projectId },
  } = useRouter()

  return (
    <>
      <Head>
        <title>API Keys</title>
      </Head>

      <PageTitle>Project API Keys</PageTitle>

      <ApiKeysCopy />

      <Tabs
        tabs={[
          { title: 'View all', href: '/[projectId]/api-keys' },
          { title: 'Create API key', href: '/[projectId]/api-keys/add' },
        ]}
      />

      <Table
        legend="Keys"
        url={`/api/v1/projects/${projectId}/apikeys/`}
        columns={['API Key Name', 'Permissions', '#Actions#']}
        expandColumn={2}
        renderEmpty="No api keys"
        renderRow={({ result, revalidate }) => (
          <ApiKeysRow
            key={result.id}
            projectId={projectId}
            apiKey={result}
            revalidate={revalidate}
          />
        )}
      />
    </>
  )
}

export default ApiKeys
