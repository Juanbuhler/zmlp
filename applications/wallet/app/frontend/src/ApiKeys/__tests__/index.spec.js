import TestRenderer from 'react-test-renderer'

import ApiKeys from '..'

describe('<ApiKeys />', () => {
  it('should render properly', () => {
    const component = TestRenderer.create(<ApiKeys />)

    expect(component.toJSON()).toMatchSnapshot()
  })
})
