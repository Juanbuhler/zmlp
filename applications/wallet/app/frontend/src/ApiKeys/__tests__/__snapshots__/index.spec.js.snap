// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ApiKeys /> should render properly while loading 1`] = `
.emotion-0 {
  font-size: 18px;
  line-height: 21px;
  font-weight: 400;
  padding-bottom: 16px;
}

.emotion-5 {
  list-style-type: none;
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 {
  padding-right: 16px;
}

.emotion-1 {
  border: 0 #b8d652 solid;
  border-bottom-width: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 8px;
  padding-bottom: 8px;
  text-transform: uppercase;
  font-size: 16px;
  line-height: 20px;
  font-weight: 700;
}

.emotion-1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-3 {
  border: 0 #b8d652 solid;
  border-bottom-width: 0;
  color: #b3b3b3;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 8px;
  padding-bottom: 8px;
  text-transform: uppercase;
  font-size: 16px;
  line-height: 20px;
  font-weight: 700;
}

.emotion-3:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #73b61c;
}

.emotion-8 {
  width: 100%;
  border-spacing: 0;
  box-shadow: 0 0 5px 0 #000000;
  white-space: nowrap;
}

.emotion-8 tr {
  background-color: #252525;
}

.emotion-8 tr:nth-of-type(2n) {
  background-color: #2e2e2e;
}

.emotion-8 tr:hover {
  background-color: #4a4a4a;
  box-shadow: 0 0 5px 0 rgba(0,0,0,0.5);
}

.emotion-8 tr:hover td {
  border: 1px solid #808080;
  border-left: 0;
  border-right: 0;
}

.emotion-8 tr:hover td:first-of-type {
  border-left: 1px solid #808080;
}

.emotion-8 tr:hover td:last-of-type {
  border-right: 1px solid #808080;
}

.emotion-8 td {
  color: #e0e0e0;
  padding: 8px 16px;
  border: 1px solid transparent;
  border-left: 0;
  border-right: 0;
}

.emotion-8 td:first-of-type {
  border-left: 1px solid transparent;
}

.emotion-8 td:last-of-type {
  border-right: 1px solid transparent;
}

.emotion-6 {
  text-align: left;
  font-size: 16px;
  line-height: 20px;
  font-weight: 500;
  color: #e0e0e0;
  background-color: #4a4a4a;
  padding: 12px 16px;
  border-bottom: 1px solid #2e2e2e;
}

.emotion-6:nth-of-type(2) {
  width: 100%;
}

.emotion-6:not(:last-child) {
  border-right: 1px solid #2e2e2e;
}

.emotion-15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px 12px;
  background-color: #808080;
  border: none;
  border-top-left-radius: 2px;
  border-bottom-left-radius: 2px;
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
}

.emotion-10:hover {
  opacity: 1;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-9 {
  color: #191919;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.emotion-11 {
  padding: 8px 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #b3b3b3;
  background-color: #000000;
}

.emotion-12 {
  padding: 8px 12px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #b3b3b3;
}

.emotion-14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px 12px;
  background-color: #808080;
  border: none;
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
  border-top-right-radius: 2px;
  border-bottom-right-radius: 2px;
}

.emotion-14:hover {
  opacity: 1;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-13 {
  color: #191919;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
}

<div>
  <h2
    className="emotion-0"
  >
    Project API Keys
  </h2>
  <nav>
    <ul
      className="emotion-5"
    >
      <li
        className="emotion-2"
      >
        <a
          className="emotion-1"
          href="/76917058-b147-4556-987a-0a0f11e46d9b/api-keys"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          View all
        </a>
      </li>
      <li
        className="emotion-2"
      >
        <a
          className="emotion-3"
          href="/76917058-b147-4556-987a-0a0f11e46d9b/api-keys/add"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Create API key
        </a>
      </li>
    </ul>
  </nav>
  <div>
     
  </div>
  <div>
    <table
      className="emotion-8"
    >
      <thead>
        <tr>
          <th
            className="emotion-6"
          >
            API Key Name
          </th>
          <th
            className="emotion-6"
          >
            Permissions
          </th>
        </tr>
      </thead>
      <tbody>
        Loading...
      </tbody>
    </table>
    <div>
       
    </div>
    <div
      className="emotion-15"
    >
      <button
        className="emotion-10"
        disabled={true}
        type="button"
      >
        <svg
          className="emotion-9"
          data-file-name="chevron.svg"
          width={16}
        />
      </button>
      <div
        className="emotion-11"
      >
        1
      </div>
      <div
        className="emotion-12"
      >
        of 
        0
      </div>
      <button
        className="emotion-14"
        disabled={true}
        type="button"
      >
        <svg
          className="emotion-13"
          data-file-name="chevron.svg"
          width={16}
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`<ApiKeys /> should render properly with api keys 1`] = `
.emotion-0 {
  font-size: 18px;
  line-height: 21px;
  font-weight: 400;
  padding-bottom: 16px;
}

.emotion-5 {
  list-style-type: none;
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 {
  padding-right: 16px;
}

.emotion-1 {
  border: 0 #b8d652 solid;
  border-bottom-width: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 8px;
  padding-bottom: 8px;
  text-transform: uppercase;
  font-size: 16px;
  line-height: 20px;
  font-weight: 700;
}

.emotion-1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-3 {
  border: 0 #b8d652 solid;
  border-bottom-width: 0;
  color: #b3b3b3;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 8px;
  padding-bottom: 8px;
  text-transform: uppercase;
  font-size: 16px;
  line-height: 20px;
  font-weight: 700;
}

.emotion-3:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #73b61c;
}

.emotion-8 {
  width: 100%;
  border-spacing: 0;
  box-shadow: 0 0 5px 0 #000000;
  white-space: nowrap;
}

.emotion-8 tr {
  background-color: #252525;
}

.emotion-8 tr:nth-of-type(2n) {
  background-color: #2e2e2e;
}

.emotion-8 tr:hover {
  background-color: #4a4a4a;
  box-shadow: 0 0 5px 0 rgba(0,0,0,0.5);
}

.emotion-8 tr:hover td {
  border: 1px solid #808080;
  border-left: 0;
  border-right: 0;
}

.emotion-8 tr:hover td:first-of-type {
  border-left: 1px solid #808080;
}

.emotion-8 tr:hover td:last-of-type {
  border-right: 1px solid #808080;
}

.emotion-8 td {
  color: #e0e0e0;
  padding: 8px 16px;
  border: 1px solid transparent;
  border-left: 0;
  border-right: 0;
}

.emotion-8 td:first-of-type {
  border-left: 1px solid transparent;
}

.emotion-8 td:last-of-type {
  border-right: 1px solid transparent;
}

.emotion-6 {
  text-align: left;
  font-size: 16px;
  line-height: 20px;
  font-weight: 500;
  color: #e0e0e0;
  background-color: #4a4a4a;
  padding: 12px 16px;
  border-bottom: 1px solid #2e2e2e;
}

.emotion-6:nth-of-type(2) {
  width: 100%;
}

.emotion-6:not(:last-child) {
  border-right: 1px solid #2e2e2e;
}

.emotion-15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px 12px;
  background-color: #808080;
  border: none;
  border-top-left-radius: 2px;
  border-bottom-left-radius: 2px;
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
}

.emotion-10:hover {
  opacity: 1;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-9 {
  color: #191919;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.emotion-11 {
  padding: 8px 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #b3b3b3;
  background-color: #000000;
}

.emotion-12 {
  padding: 8px 12px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #b3b3b3;
}

.emotion-14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px 12px;
  background-color: #808080;
  border: none;
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
  border-top-right-radius: 2px;
  border-bottom-right-radius: 2px;
}

.emotion-14:hover {
  opacity: 1;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-13 {
  color: #191919;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
}

<div>
  <h2
    className="emotion-0"
  >
    Project API Keys
  </h2>
  <nav>
    <ul
      className="emotion-5"
    >
      <li
        className="emotion-2"
      >
        <a
          className="emotion-1"
          href="/76917058-b147-4556-987a-0a0f11e46d9b/api-keys"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          View all
        </a>
      </li>
      <li
        className="emotion-2"
      >
        <a
          className="emotion-3"
          href="/76917058-b147-4556-987a-0a0f11e46d9b/api-keys/add"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Create API key
        </a>
      </li>
    </ul>
  </nav>
  <div>
     
  </div>
  <div>
    <table
      className="emotion-8"
    >
      <thead>
        <tr>
          <th
            className="emotion-6"
          >
            API Key Name
          </th>
          <th
            className="emotion-6"
          >
            Permissions
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            admin-key
          </td>
          <td>
             Super Admin,  Project Admin,  Assets Read,  Assets Import
          </td>
        </tr>
      </tbody>
    </table>
    <div>
       
    </div>
    <div
      className="emotion-15"
    >
      <button
        className="emotion-10"
        disabled={true}
        type="button"
      >
        <svg
          className="emotion-9"
          data-file-name="chevron.svg"
          width={16}
        />
      </button>
      <div
        className="emotion-11"
      >
        1
      </div>
      <div
        className="emotion-12"
      >
        of 
        1
      </div>
      <button
        className="emotion-14"
        disabled={true}
        type="button"
      >
        <svg
          className="emotion-13"
          data-file-name="chevron.svg"
          width={16}
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`<ApiKeys /> should render properly with no api keys 1`] = `
.emotion-0 {
  font-size: 18px;
  line-height: 21px;
  font-weight: 400;
  padding-bottom: 16px;
}

.emotion-5 {
  list-style-type: none;
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 {
  padding-right: 16px;
}

.emotion-1 {
  border: 0 #b8d652 solid;
  border-bottom-width: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 8px;
  padding-bottom: 8px;
  text-transform: uppercase;
  font-size: 16px;
  line-height: 20px;
  font-weight: 700;
}

.emotion-1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-3 {
  border: 0 #b8d652 solid;
  border-bottom-width: 0;
  color: #b3b3b3;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 8px;
  padding-bottom: 8px;
  text-transform: uppercase;
  font-size: 16px;
  line-height: 20px;
  font-weight: 700;
}

.emotion-3:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #73b61c;
}

.emotion-10 {
  width: 100%;
  border-spacing: 0;
  box-shadow: 0 0 5px 0 #000000;
  white-space: nowrap;
}

.emotion-10 tr {
  background-color: #252525;
}

.emotion-10 tr:nth-of-type(2n) {
  background-color: #2e2e2e;
}

.emotion-10 tr:hover {
  background-color: #4a4a4a;
  box-shadow: 0 0 5px 0 rgba(0,0,0,0.5);
}

.emotion-10 tr:hover td {
  border: 1px solid #808080;
  border-left: 0;
  border-right: 0;
}

.emotion-10 tr:hover td:first-of-type {
  border-left: 1px solid #808080;
}

.emotion-10 tr:hover td:last-of-type {
  border-right: 1px solid #808080;
}

.emotion-10 td {
  color: #e0e0e0;
  padding: 8px 16px;
  border: 1px solid transparent;
  border-left: 0;
  border-right: 0;
}

.emotion-10 td:first-of-type {
  border-left: 1px solid transparent;
}

.emotion-10 td:last-of-type {
  border-right: 1px solid transparent;
}

.emotion-6 {
  text-align: left;
  font-size: 16px;
  line-height: 20px;
  font-weight: 500;
  color: #e0e0e0;
  background-color: #4a4a4a;
  padding: 12px 16px;
  border-bottom: 1px solid #2e2e2e;
}

.emotion-6:nth-of-type(2) {
  width: 100%;
}

.emotion-6:not(:last-child) {
  border-right: 1px solid #2e2e2e;
}

.emotion-17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px 12px;
  background-color: #808080;
  border: none;
  border-top-left-radius: 2px;
  border-bottom-left-radius: 2px;
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
}

.emotion-12:hover {
  opacity: 1;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-11 {
  color: #191919;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.emotion-13 {
  padding: 8px 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #b3b3b3;
  background-color: #000000;
}

.emotion-14 {
  padding: 8px 12px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #b3b3b3;
}

.emotion-16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px 12px;
  background-color: #808080;
  border: none;
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
  border-top-right-radius: 2px;
  border-bottom-right-radius: 2px;
}

.emotion-16:hover {
  opacity: 1;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-15 {
  color: #191919;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
}

.emotion-9 {
  pointer-events: none;
}

.emotion-8 {
  height: 300px;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  color: red;
  font-size: 16px;
  line-height: 20px;
}

<div>
  <h2
    className="emotion-0"
  >
    Project API Keys
  </h2>
  <nav>
    <ul
      className="emotion-5"
    >
      <li
        className="emotion-2"
      >
        <a
          className="emotion-1"
          href="/76917058-b147-4556-987a-0a0f11e46d9b/api-keys"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          View all
        </a>
      </li>
      <li
        className="emotion-2"
      >
        <a
          className="emotion-3"
          href="/76917058-b147-4556-987a-0a0f11e46d9b/api-keys/add"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Create API key
        </a>
      </li>
    </ul>
  </nav>
  <div>
     
  </div>
  <div>
    <table
      className="emotion-10"
    >
      <thead>
        <tr>
          <th
            className="emotion-6"
          >
            API Key Name
          </th>
          <th
            className="emotion-6"
          >
            Permissions
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          aria-label="Empty table"
          className="emotion-9"
        >
          <td
            colSpan={2}
          >
            <div
              className="emotion-8"
            >
              No api keys
            </div>
          </td>
        </tr>
      </tbody>
    </table>
    <div>
       
    </div>
    <div
      className="emotion-17"
    >
      <button
        className="emotion-12"
        disabled={true}
        type="button"
      >
        <svg
          className="emotion-11"
          data-file-name="chevron.svg"
          width={16}
        />
      </button>
      <div
        className="emotion-13"
      >
        1
      </div>
      <div
        className="emotion-14"
      >
        of 
        0
      </div>
      <button
        className="emotion-16"
        disabled={true}
        type="button"
      >
        <svg
          className="emotion-15"
          data-file-name="chevron.svg"
          width={16}
        />
      </button>
    </div>
  </div>
</div>
`;
